scripts/generate-pg-include-guard.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -euo pipefail
+
+CPP_ONLY_FILE=$1
+PG_INCLUDE_DIR=$2
+
+OUTPUT_DIR=$(dirname "${CPP_ONLY_FILE}")
+
+mkdir -p "${OUTPUT_DIR}"
+
+echo "Will generate: $(pwd)/${CPP_ONLY_FILE}"
+
+echo "// Auto-generated file, run 'make update_cpp_guard' to update" > "${CPP_ONLY_FILE}"
+echo "#if \\" >> "${CPP_ONLY_FILE}"
+
+
+grep -R '_H$' "${PG_INCLUDE_DIR}" | grep define | cut -d: -f 2 | cut -d \  -f 2 | sort | while read -r m;
```

Comment:
I think this is a overkill. I think failing on `POSTGRES_H` being defined enough for all practical cases that we'll run into.

Reply:
Well I went there cause I found a few files that didn't include `postgres.h`, so I was concerned for false positives. But yeah only checking `POSTGRES_H` would certainly be much simpler. Let's chat about it tomorrow.

Reply:
Ok, after sleeping on it I think I agree that the small risk is worth the simplicity. I'm cleaned up the PR.